fvm/handler/programs.go

```diff
@@ -22,19 +22,21 @@ type stackEntry struct {
 // naturally, making cleanup method essentially no-op. But if something goes wrong, all nested
 // views must be merged in order to make sure they are recorded
 type ProgramsHandler struct {
-	masterState  *state.StateHolder
-	viewsStack   []stackEntry
-	Programs     *programs.Programs
-	initialState *state.State
+	masterState    *state.StateHolder
+	viewsStack     []stackEntry
+	Programs       *programs.Programs
+	LoadedPrograms map[common.LocationID]interface{}
```

Comment:
⚠️ this is not safe and might cause execution fork, this can only happen through the state holder

Reply:
I need to think more about it